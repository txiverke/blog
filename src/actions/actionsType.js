// @flow

export const LOAD_STATS_REQUEST = 'LOAD_STATS_REQUEST'
export const LOAD_STATS_SUCCESS = 'LOAD_STATS_SUCCESS'
export const LOAD_STATS_FAILURE = 'LOAD_STATS_FAILURE'

export const LOG_IN_USER_REQUEST = 'LOG_IN_USER_REQUEST'
export const LOG_IN_USER_SUCCESS = 'LOG_IN_USER_SUCCESS'
export const LOG_IN_USER_FAILURE = 'LOG_IN_USER_FAILURE'

export const IS_AUTHENTICATED_REQUEST = 'IS_AUTHENTICATE_REQUEST'
export const IS_AUTHENTICATED_SUCCESS = 'IS_AUTHENTICATE_SUCCESS'
export const IS_AUTHENTICATED_FAILURE = 'IS_AUTHENTICATE_FAILURE'

export const LOAD_USER_DATA_REQUEST = 'LOAD_USER_DATA_REQUEST'
export const LOAD_USER_DATA_SUCCESS = 'LOAD_USER_DATA_SUCCESS'
export const LOAD_USER_DATA_FAILURE = 'LOAD_USER_DATA_FAILURE'

export const UPDATE_USER_DATA_REQUEST = 'UPDATE_USER_DATA_REQUEST'
export const UPDATE_USER_DATA_SUCCESS = 'UPDATE_USER_DATA_SUCCESS'
export const UPDATE_USER_DATA_FAILURE = 'UPDATE_USER_DATA_FAILURE'

export const LOG_OUT_USER_REQUEST = 'LOG_OUT_USER_REQUEST'
export const LOG_OUT_USER_SUCCESS = 'LOG_OUT_USER_SUCCESS'
export const LOG_OUT_USER_FAILURE = 'LOG_OUT_USER_FAILURE'

export const LOAD_POST_DATA_REQUEST = 'LOAD_POST_DATA_REQUEST'
export const LOAD_POST_DATA_SUCCESS = 'LOAD_POST_DATA_SUCCESS'
export const LOAD_POST_DATA_FAILURE = 'LOAD_POST_DATA_FAILURE'

export const LOAD_POST_ITEM_REQUEST = 'LOAD_POST_ITEM_REQUEST'
export const LOAD_POST_ITEM_SUCCESS = 'LOAD_POST_ITEM_SUCCESS'
export const LOAD_POST_ITEM_FAILURE = 'LOAD_POST_ITEM_FAILURE'
export const UPDATE_POST_ITEM_SUCCESS = 'UPDATE_POST_ITEM_SUCCESS'

export const CREATE_POST_DATA_REQUEST = 'CREATE_POST_DATA_REQUEST'
export const CREATE_POST_DATA_SUCCESS = 'CREATE_POST_DATA_SUCCESS'
export const CREATE_POST_DATA_FAILURE = 'CREATE_POST_DATA_FAILURE'

export const UPDATE_POST_DATA_REQUEST = 'UPDATE_POST_DATA_REQUEST'
export const UPDATE_POST_DATA_SUCCESS = 'UPDATE_POST_DATA_SUCCESS'
export const UPDATE_POST_DATA_FAILURE = 'UPDATE_POST_DATA_FAILURE'

export const REMOVE_POST_DATA_REQUEST = 'REMOVE_POST_DATA_REQUEST'
export const REMOVE_POST_DATA_SUCCESS = 'REMOVE_POST_DATA_SUCCESS'
export const REMOVE_POST_DATA_FAILURE = 'REMOVE_POST_DATA_FAILURE'
