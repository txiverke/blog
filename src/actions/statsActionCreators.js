// @flow

import * as ACTION from './actionsType'
import config from '../config'
import { handleToken } from '../utils/helpers'

/** LOAD APP STATISTICS */
export const loadStatsRequest = () => ({ type: ACTION.LOAD_STATS_REQUEST })
export const loadStatsSuccess = (payload: Object) => ({ type: ACTION.LOAD_STATS_SUCCESS, payload })
export const loadStatsFailure = () => ({ type: ACTION.LOAD_STATS_FAILURE })

export const loadStats = () => 
  async (dispatch: Function) => {
    const headers = new Headers({ 'Content-Type': 'application/json' })
    const options = { 
      method: 'GET',
      headers,
      mode: 'cors',
      cache: 'default',
    }

    dispatch(loadStatsRequest())

    try {
      const promise = await fetch(`${config.api.url}/statistic`, options)
      const data = await promise.json()
      return dispatch(loadStatsSuccess(data))
    } catch (err) {
      dispatch(loadStatsFailure())
    }
  } 




