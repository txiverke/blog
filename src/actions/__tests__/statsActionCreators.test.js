import request from 'supertest-as-promised'
import * as action from '../statsActionCreators'
import * as type from '../actionsType'

describe('statsActionCreators', () => {
  it('should return LOAD_STATS_REQUEST action', () => {
    expect(action.loadStatsRequest()).toMatchSnapshot()
  })

  it('should return LOAD_STATS_SUCCESS action', () => {
    const result = { testing: true }
    expect(action.loadStatsSuccess(result)).toMatchSnapshot()
  })

  it('should return LOAD_STATS_FAILURE action', () => {
    expect(action.loadStatsFailure()).toMatchSnapshot()
  })

})