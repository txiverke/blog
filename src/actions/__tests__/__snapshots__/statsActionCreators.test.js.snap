// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statsActionCreators should return LOAD_STATS_FAILURE action 1`] = `
Object {
  "type": "LOAD_STATS_FAILURE",
}
`;

exports[`statsActionCreators should return LOAD_STATS_REQUEST action 1`] = `
Object {
  "type": "LOAD_STATS_REQUEST",
}
`;

exports[`statsActionCreators should return LOAD_STATS_SUCCESS action 1`] = `
Object {
  "payload": Object {
    "testing": true,
  },
  "type": "LOAD_STATS_SUCCESS",
}
`;
